base-template.sh

```diff
@@ -522,15 +600,36 @@ update_shared_hooks_if_appropriate() {
     [ "$HOOK_NAME" = "post-checkout" ] && [ "$1" = "$GIT_NULL_REF" ] && RUN_UPDATE="true"
 
     if [ "$RUN_UPDATE" = "true" ]; then
+
         # split on comma and newline
-        IFS="$IFS_COMMA_NEWLINE"
+        IFS_TMP="$IFS_COMMA_NEWLINE"
+        [ "$SHARED_HOOKS_TYPE" = "--shared" ] && IFS_TMP="$IFS_NEWLINE"
+        IFS="$IFS_TMP"
 
         for SHARED_REPO in $SHARED_REPOS_LIST; do
             unset IFS
-            mkdir -p "$INSTALL_DIR/shared"
 
             set_shared_root "$SHARED_REPO"
 
+            if [ "$SHARED_REPO_IS_CLONED" != "true" ]; then
+                # Non-cloned roots are ignored
+                continue
+            elif [ "$SHARED_HOOKS_TYPE" = "--shared" ] &&
+                [ "$SHARED_REPO_IS_LOCAL" = "true" ]; then
+                echo "! Warning: Shared hooks in \`.githooks/.shared\` contain a local path" >&2
+                echo "  \`$SHARED_REPO\`" >&2
+                echo "  which is forbidden. Update will be skipped." >&2
+                echo "" >&2
+                echo "  You can only have local paths for shared hooks defined" >&2
+                echo "  in the local or global Git configuration." >&2
+                echo "" >&2
+                echo "  This can be achieved by running" >&2
+                echo "    \$ git hooks shared add [--local|--global] \"$SHARED_REPO\"" >&2
+                echo "  and deleting it from the \`.shared\` file manually by" >&2
+                echo "    \$ git hooks shared remove --shared \"$SHARED_REPO\"" >&2
+                continue
+            fi
+
             if [ -d "$SHARED_ROOT/.git" ]; then
                 echo "* Updating shared hooks from: $SHARED_REPO"
                 PULL_OUTPUT=$(git -C "$SHARED_ROOT" --work-tree="$SHARED_ROOT" --git-dir="$SHARED_ROOT/.git" -c core.hooksPath=/dev/null pull 2>&1)
```

Comment:
Ah nice, didnt kow this exists. I will introduce this to all clone stuff we have in the code. leave that open.

Reply:
Couldnt apply this, I had problem with depth=1 in `pull` -> `cannot merge unrelated history..." ...
I leave it for now...

Reply:
Ah, right. Pull probably does a fetch with `--depth` and then tries to merge into the local branch, which it can't for lack of path between the two commits.

One way around this would be to do a `fetch` followed by a `reset --keep FETCH_HEAD` or something like that, but that is probably more fragile to get right, so maybe just doing a shallow clone and then normal is fine anyway.

Reply:
Jep I leave it that way, also because the clone/pull of the release its absoultely necessary to have the history when an update was made...